src/status_im/subs/wallet/wallet_connect.cljs

```diff
@@ -58,6 +60,26 @@
       (some #(string/includes? % address) accounts))
     sessions)))
 
+(rf/reg-sub
+ :wallet-connect/sessions-for-current-account-and-networks
+ :<- [:wallet-connect/sessions-for-current-account]
+ :<- [:profile/test-networks-enabled?]
+ (fn [[sessions testnet-mode?]]
+   (filter
+    (fn [{:keys [chains]}]
+      (let [comparison-fn (if testnet-mode? (partial not set/subset?) set/subset?)
+            chain-numbers (set (map (fn [chain]
+                                      (-> chain
+                                          (string/split ":")
+                                          second
+                                          js/parseInt))
+                                    chains))]
+        ;; If network is testnet, we ensure that the required chains for that session are not a
+        ;; subset of mainnet chain ids. If network is mainnet, we ensure that the required chains
+        ;; for that session are a subset of mainnet chain ids
+        (comparison-fn chain-numbers constants/mainnet-chain-ids)))
+    sessions)))
```

Comment:
Maybe I got it wrong but it seems like you use `constants/mainnet-chain-ids` for Mainnet chain ids, and basically treat all the other chain ids as Testnet chain ids.
But is it correct?

Reply:
Since if there are no mainnet chain ids in the session, I guess we could assume there would be at least one testnet chain id there, which would make it a "testnet session".